File: wpimath/src/test/java/edu/wpi/first/math/controller/proto/SimpleMotorFeedforwardProtoTest.java
// Copyright (c) FIRST and other WPILib contributors.
// Open Source Software; you can modify and/or share it under the terms of
// the WPILib BSD license file in the root directory of this project.

package edu.wpi.first.math.controller.proto;

import static org.junit.jupiter.api.Assertions.assertEquals;

import edu.wpi.first.math.controller.SimpleMotorFeedforward;
import edu.wpi.first.math.proto.Controller.ProtobufSimpleMotorFeedforward;
import edu.wpi.first.wpilibj.ProtoTestBase;

@SuppressWarnings("PMD.TestClassWithoutTestCases")
class SimpleMotorFeedforwardProtoTest
    extends ProtoTestBase<SimpleMotorFeedforward, ProtobufSimpleMotorFeedforward> {
  SimpleMotorFeedforwardProtoTest() {
    super(new SimpleMotorFeedforward(0.4, 4.0, 0.7), SimpleMotorFeedforward.proto);
  }

  @Override
  public void checkEquals(SimpleMotorFeedforward testData, SimpleMotorFeedforward data) {
    assertEquals(testData.ks, data.ks);
    assertEquals(testData.kv, data.kv);
    assertEquals(testData.ka, data.ka);
  }
}

File: wpimath/src/test/native/cpp/controller/proto/SimpleMotorFeedforwardProtoTest.cpp
// Copyright (c) FIRST and other WPILib contributors.
// Open Source Software; you can modify and/or share it under the terms of
// the WPILib BSD license file in the root directory of this project.

#include <gtest/gtest.h>

#include "../../ProtoTestBase.h"
#include "frc/controller/SimpleMotorFeedforward.h"

using namespace frc;

struct SimpleMotorFeedforwardProtoTestData {
  using Type = SimpleMotorFeedforward<units::meters>;

  inline static const Type kTestData = {units::volt_t{0.4},
                                        units::volt_t{4.0} / 1_mps,
                                        units::volt_t{0.7} / 1_mps_sq};

  static void CheckEq(const Type& testData, const Type& data) {
    EXPECT_EQ(testData.kS.value(), data.kS.value());
    EXPECT_EQ(testData.kV.value(), data.kV.value());
    EXPECT_EQ(testData.kA.value(), data.kA.value());
  }
};

INSTANTIATE_TYPED_TEST_SUITE_P(SimpleMotorFeedforwardMeters, ProtoTest,
                               SimpleMotorFeedforwardProtoTestData);
